lib/HLSL/DxilValueCache.cpp

```diff
@@ -170,7 +192,15 @@ Value *DxilValueCache::ProcessAndSimpilfy_Br(Instruction *I, DominatorTree *DT)
   return nullptr;
 }
 
-
+Value *DxilValueCache::ProcessAndSimpilfy_Load(Instruction *I, DominatorTree *DT) {
+  LoadInst *LI = cast<LoadInst>(I);
+  Value *V = OptionallyGetValue(LI->getPointerOperand());
+  if (Constant *ConstPtr = dyn_cast<Constant>(V)) {
+    const DataLayout &DL = I->getModule()->getDataLayout();
+    return llvm::ConstantFoldLoadFromConstPtr(ConstPtr, DL);
```

Comment:
Watch out for initialized external constant globals representing constant buffer values, because I swear I've seen them used with zeroinitializer, which isn't technically correct, but would cause bugs with this.